### PR TITLE
[Promotion] Use proportional distribution in order percentage discount

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
We should add more scenarios here. The order percentage discount should also take into account shipping fee.

File: src/Sylius/Component/Core/Promotion/Action/PercentageDiscountAction.php [hunk 1]
```diff
@@ -63,7 +66,12 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
             return;
         }
 
-        $splitPromotion = $this->distributor->distribute($promotionAmount, $subject->countItems());
+        $itemsTotal = [];
+        foreach ($subject->getItems() as $orderItem) {
+            $itemsTotal[] = $orderItem->getTotal();
+        }
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Promotion/Action/PercentageDiscountAction.php]: There is `getItemsTotal()` method on `Order`

Comment (pamil) [inline src/Sylius/Component/Core/Promotion/Action/PercentageDiscountAction.php]: It returns the sum of items totals, not the array with items totals.

**Comment by pamil** [comment]
@michalmarcinkowski there's "Receiving percentage discount does not affect the shipping fee" scenario, so we should rather discuss on it.

**Comment by michalmarcinkowski** [comment]
@pamil that's great! I recall we simplified that, so you should configure shipping percentage discount separately. Good job then! 👍

**Pull request merged**